dts/arm/atmel/samd.dtsi

```diff
@@ -0,0 +1,137 @@
+/*
```

Comment:
nit: Atmel now have a SAMD51 series which seems to be a SAM0 with a Cortex M4.  I wouldn't worry about it now, but we may need to rename this in the future.

Reply:
We probably shouldn't be doing it. SAMD family has the following series: D09, D10, D11, D20, D21, DA1, HA1 and they are all quite different. E.g several D10 peripherals are located at different addresses than D21/D20. While it's true that D20 and D21 look fairly similar if we try to maintain one common file it will very quickly get messy. Even at present this file contains an entry for USB0 peripheral which is missing on D20 chip.

Let's maybe keep one dtsi file per series and only ensure that they are easy to compare with diff.